File: 05/b.c
// gcc b.c -o b -lcrypto && ./b

#include <openssl/md5.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main()
{
    char prefix[9] = "uqwqemis";    
    char in[33];
    unsigned char outb[16];
    char out[33];
    int i = 0;
    int c = 0;
    char res[9] = "--------";

    for (;;) {
        sprintf(in, "%s%d", prefix, i);

        MD5(in, strlen(in), outb);

        for (int j = 0; j < 16; j++)
            sprintf(out + j * 2, "%02x", outb[j]);            
        
        int valid =
            out[0] == '0' &&
            out[1] == '0' &&
            out[2] == '0' &&
            out[3] == '0' &&
            out[4] == '0' &&

            out[5] >= '0' &&
            out[5] <= '8';

        if (valid) {
            char chr[2] = { out[5] };
            int index = atoi(chr);

            if (res[index] == '-') {
                res[index] = out[6];
                c++;
            }

            printf("%s\n", res);
        }

        if (c > 7)
            break;

        i++;
    }

    printf("%s\n", res);
}

File: 05/b.zig
const std = @import("std");
const stdout = &std.io.getStdOut().outStream().stream;
const Md5 = std.crypto.Md5;

pub fn main() !void {
    var prefix = [_]u8{ 'u', 'q', 'w', 'q', 'e', 'm', 'i', 's',  };
    var in = [_]u8{0} ** 33;
    var outb = [_]u8{0} ** 16;
    var out = [_]u8{0} ** 33;
    var i: u32 = 0;
    var c: u32 = 0;
    var res = [_]u8{ '-', '-', '-', '-', '-', '-', '-', '-' };

    while (true) {
        var buf = try std.fmt.bufPrint(in[0..], "{}{}", .{prefix, i});

        var size: usize = 0;
        for (in[0..]) |byte, index| {
            if (byte == 0) {
                size = index;
                break;
            }
        }

        Md5.hash(in[0..size], outb[0..]);

        var buf2 = try std.fmt.bufPrint(out[0..], "{x}", .{outb});

        var valid = 
            out[0] == '0' and
            out[1] == '0' and
            out[2] == '0' and
            out[3] == '0' and
            out[4] == '0' and

            out[5] >= '0' and
            out[5] <= '8';

        if (valid) {
            var index: u32 = out[5] - '0';
            
            if (res[index] == '-') {
                res[index] = out[6];
                c += 1;
            }
            
            try stdout.print("{}\n", .{res});
        }

        if (c > 7)
            break;

        i += 1;
    }

    try stdout.print("{}\n", .{res});
}
